torchao/dtypes/uintx/gemlite_layout.py

```diff
@@ -297,14 +297,31 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
         if func is aten.slice.Tensor:
```

Comment:
can we update this class to use the @dispatch pattern

Reply:
sure, can I do this in a separate PR?